source/Timer.cpp

```diff
@@ -0,0 +1,253 @@
+/* Timer.cpp
+Copyright (c) 2023 by Timothy Collett
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Timer.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "GameData.h"
+#include "Logger.h"
+#include "Planet.h"
+#include "PlayerInfo.h"
+#include "Random.h"
+#include "Ship.h"
+#include "StellarObject.h"
+#include "UI.h"
+
+using namespace std;
+
+Timer::Timer(const DataNode &node, Mission *mission)
+{
+	Load(node, mission);
+}
+
+void Timer::Load(const DataNode &node, Mission *mission)
+{
+	if(node.Size() > 1)
+		name = node.Token(1);
+	if(node.Size() > 2)
+		base = static_cast<int64_t>(node.Value(2));
+	if(node.Size() > 3)
+		rand = static_cast<uint32_t>(node.Value(3));
+
+	this->mission = mission;
+
+	timeToWait = base + Random::Int(rand);
+
+	for(const DataNode &child : node)
+	{
+		if(child.Token(0) == "idle")
+			requireIdle = true;
+		else if(child.Token(0) == "system" && child.Size() > 1)
```

Comment:
Not sure but `"source"` sounds better imo? idk though

Reply:
Hm. "Source" seems to imply that it's coming _from_ the system where you have to wait. I would almost say "destination" fits better, but in both cases I think (as with the trigger, below) it is better to have it be something distinct from the specifiers for the mission itself, to reduce confusion. 

I suppose we could change it to something even more descriptive, like `wait in`? Though that has the disadvantage of requiring quoting...

On the whole, I'm inclined to stick with `system` unless you really think it should be changed.